tools/asset-sync/assets.ps1

```diff
@@ -0,0 +1,521 @@
+#>
+Function Is-AssetsRepo-Initialized {
+    param(
+        [PSCustomObject] $Config
+    )
+
+    $result = $false
+    $assetRepoLocation = Resolve-AssetRepo-Location -Config $Config
+
+    try {
+        Push-Location $assetRepoLocation
+        $gitLocation = Join-Path $assetRepoLocation ".git"
+
+        if (Test-Path $gitLocation){
+            $result = $true
+        }
+        else{
+            $result = $false
+        }
+    }
+    catch {
+        Write-Error $_
+        $result = $false
+    }
+    finally {
+        Pop-Location
+    }
+
+    return $result
+}
+
+
+<#
+.SYNOPSIS
+Given a configuration, determine which paths must be added to the sparse checkout of the assets repo.
+
+.PARAMETER Config
+A PSCustomObject that contains an auto-parsed assets.json content.
+
+#>
+Function Resolve-CheckoutPaths {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config
+    )
+    $assetsJsonPath = $Config.AssetsJsonRelativeLocation
+
+    $assetsJsonFolder = Split-Path $Config.AssetsJsonRelativeLocation
+
+    if(!$assetsJsonFolder -or $assetsJsonFolder -in @(".", "./", ".\"))
+    {
+        return $Config.AssetsRepoPrefixPath.Replace("`\", "/")
+    }
+    else {
+        $result = (Join-Path $Config.AssetsRepoPrefixPath $assetsJsonFolder).Replace("`\", "/")
+        return $result
+    }
+}
+
+<#
+.SYNOPSIS
+Given a configuration, determine the _current_ target path.
+
+.DESCRIPTION 
+Determines the presence of a branch on the git repo. If the relevant auto/<service> branch does not exist, we should use main.
+#>
+Function Resolve-TargetBranch {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config
+    )
+    $assetRepo = Resolve-AssetRepo-Location -Config $Config
+    $branch = "main"
+    try {
+        Push-Location $assetRepo
+        
+        $latestCommit = git rev-parse "origin/$($Config.AssetsRepoBranch)"
+
+        if($LASTEXITCODE -eq 0) {
+            $branch = $Config.AssetsRepoBranch
+        }
+    }
+    finally {
+        Pop-Location
+    }
+
+    return $branch
+}
+
+<#
+.SYNOPSIS
+Initializes a recordings repo based on an assets.json file. 
+
+.DESCRIPTION
+This Function will NOT re-initialize a repo if it discovers the repo already ready to go.
+
+.PARAMETER Config
+A PSCustomObject that contains an auto-parsed assets.json content from Resolve-AssetsJson.
+
+.PARAMETER ForceReinitialize
+Should this assets repo be renewed regardless of current status?
+#>
+Function Initialize-AssetsRepo {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config,
+        [Parameter(Mandatory=$false)]
+        [boolean] $ForceReinitialize = $false
+    )
+    $assetRepo = Resolve-AssetRepo-Location -Config $Config
+    $initialized = Is-AssetsRepo-Initialized -Config $Config
+    $workCompleted = $false
+
+    if ($ForceReinitialize)
+    {
+        Remove-Item -Force -R "$assetRepo/*"
```

Comment:
I would be extra careful here to check that `$assetRepo` is not empty string...

Reply:
Oof. Good call.